MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/snapshotter/MapSnapshotter.java

```diff
@@ -117,23 +124,20 @@ public Options(int width, int height) {
     }
 
     /**
-     * @param uri The style URI to use
+     * Set a style builder to snapshotter, the contents in builder like layers/sources/images will be applied
+     * to snapshotter.
+     *
+     * @param builder The builder will applied to snapshotter
      * @return the mutated {@link Options}
      */
     @NonNull
-    public Options withStyle(String uri) {
-      this.styleUri = uri;
+    public Options withStyleBuilder(Style.Builder builder) {
+      this.builder = builder;
       return this;
     }
 
-    /**
-     * @param styleJson The style json to use
-     * @return the mutated {@link Options}
-     */
-    @NonNull
-    public Options withStyleJson(String styleJson) {
```

Comment:
We cannot remove these methods. We are stuck with sementic versioning, We need to deprecate these methods and make sure the API doesn't break.